test/ectest.c

```diff
@@ -188,1185 +146,1027 @@ static void group_order_tests(EC_GROUP *group)
-        ABORT;
+    if (!TEST_ptr(ctx = BN_CTX_new())
+        || !TEST_ptr(p = BN_new()) || !TEST_ptr(a = BN_new())
+        || !TEST_ptr(b = BN_new())
+        || !TEST_true(BN_hex2bn(&p, "17"))
+        || !TEST_true(BN_hex2bn(&a, "1"))
+        || !TEST_true(BN_hex2bn(&b, "1"))
+        /*
+         * applications should use EC_GROUP_new_curve_GFp so
+         * that the library gets to choose the EC_METHOD
+         */
+        || !TEST_ptr(group = EC_GROUP_new(EC_GFp_mont_method()))
+        || !TEST_true(EC_GROUP_set_curve_GFp(group, p, a, b, ctx))
+        || !TEST_ptr(tmp = EC_GROUP_new(EC_GROUP_method_of(group)))
+        || !TEST_true(EC_GROUP_copy(tmp, group)))
+        goto err;
+    EC_GROUP_free(group);
+    group = tmp;
+    tmp = NULL;
+
+    if (!TEST_true(EC_GROUP_get_curve_GFp(group, p, a, b, ctx)))
+        goto err;
+
+    BIO_printf(bio_out,
+            "Curve defined by Weierstrass equation\n"
+            "     y^2 = x^3 + a*x + b  (mod 0x");
+    BN_print(bio_out, p);
+    BIO_printf(bio_out, ")\n     a = 0x");
+    BN_print(bio_out, a);
+    BIO_printf(bio_out, "\n     b = 0x");
+    BN_print(bio_out, b);
+    BIO_printf(bio_out, "\n");
 
     buf[0] = 0;
-    if (!EC_POINT_oct2point(group, Q, buf, 1, ctx))
-        ABORT;
-
-    if (!EC_POINT_add(group, P, P, Q, ctx))
-        ABORT;
-    if (!EC_POINT_is_at_infinity(group, P))
-        ABORT;
-
-    x = BN_new();
-    y = BN_new();
-    z = BN_new();
-    yplusone = BN_new();
-    if (x == NULL || y == NULL || z == NULL || yplusone == NULL)
-        ABORT;
-
-    if (!BN_hex2bn(&x, "D"))
-        ABORT;
-    if (!EC_POINT_set_compressed_coordinates_GFp(group, Q, x, 1, ctx))
-        ABORT;
-    if (EC_POINT_is_on_curve(group, Q, ctx) <= 0) {
-        if (!EC_POINT_get_affine_coordinates_GFp(group, Q, x, y, ctx))
-            ABORT;
-        fprintf(stderr, "Point is not on curve: x = 0x");
+    if (!TEST_ptr(P = EC_POINT_new(group)) || !TEST_ptr(Q = EC_POINT_new(group))
+        || !TEST_ptr(R = EC_POINT_new(group))
+        || !TEST_true(EC_POINT_set_to_infinity(group, P))
+        || !TEST_true(EC_POINT_is_at_infinity(group, P))
+        || !TEST_true(EC_POINT_oct2point(group, Q, buf, 1, ctx))
+        || !TEST_true(EC_POINT_add(group, P, P, Q, ctx))
+        || !TEST_true(EC_POINT_is_at_infinity(group, P))
+        || !TEST_ptr(x = BN_new()) || !TEST_ptr(y = BN_new())
+        || !TEST_ptr(z = BN_new()) || !TEST_ptr(yplusone = BN_new())
+        || !TEST_true(BN_hex2bn(&x, "D"))
+        || !TEST_true(EC_POINT_set_compressed_coordinates_GFp(group, Q, x, 1,
+                                                              ctx)))
+        goto err;
+
+    if (!TEST_int_gt(EC_POINT_is_on_curve(group, Q, ctx), 0)) {
+        if (!TEST_true(EC_POINT_get_affine_coordinates_GFp(group, Q, x, y,
+                                                           ctx)))
+            goto err;
+        BIO_printf(bio_err, "Point is not on curve: x = 0x");
```

Comment:
I like the first and last ideas.  @levitte and @dot-asm since BIO_printf is now separate from stdio, what do you think of adding a %B formatter for BIGNUM's?